src/mesh/mesh.cpp

```diff
@@ -553,14 +555,11 @@ Mesh::Mesh(ParameterInput *pin, ApplicationInput *app_in, RestartReader &rr,
   // initialize
   loclist = std::vector<LogicalLocation>(nbtotal);
 
-  const auto blockSize = rr.GetAttrVec<int>("Info", "MeshBlockSize");
-  const auto includesGhost = rr.GetAttr<int>("Info", "IncludesGhost");
-  const auto nGhost = rr.GetAttr<int>("Info", "NGhost");
-
   for (auto &dir : {X1DIR, X2DIR, X3DIR}) {
     block_size.xrat(dir) = mesh_size.xrat(dir);
-    block_size.nx(dir) =
-        blockSize[dir - 1] - (blockSize[dir - 1] > 1) * includesGhost * 2 * nGhost;
+    block_size.nx(dir) = mesh_info.block_size[dir - 1] -
+                         (mesh_info.block_size[dir - 1] > 1) * mesh_info.includes_ghost *
+                             2 * mesh_info.n_ghost;
```

Comment:
With different topological elements, this information will be less useful than it used to be. But I'm not sure where all the dependencies are on this.

Reply:
Yeah, I was also wondering about this, but decided to keep this PR on a refactor and not cleanup.